### YamlMime:ManagedReference
items:
- uid: System.Threading.ThreadStateException
  id: ThreadStateException
  children:
  - System.Threading.ThreadStateException.#ctor
  - System.Threading.ThreadStateException.#ctor(System.String)
  - System.Threading.ThreadStateException.#ctor(System.Runtime.Serialization.SerializationInfo,System.Runtime.Serialization.StreamingContext)
  - System.Threading.ThreadStateException.#ctor(System.String,System.Exception)
  langs:
  - csharp
  name: ThreadStateException
  nameWithType: ThreadStateException
  fullName: System.Threading.ThreadStateException
  type: Class
  summary: "Undantag som utlöses när en <xref href=&quot;System.Threading.Thread&quot;> </xref> är i ett ogiltigt <xref:System.Threading.Thread.ThreadState*>för metodanropet.</xref:System.Threading.Thread.ThreadState*>"
  remarks: "När en tråd har skapats kan den är i minst en av de <xref:System.Threading.ThreadState>tillstånd tills den avslutas.</xref:System.Threading.ThreadState> `ThreadStateException`genereras av metoder som inte kan utföra den begärda åtgärden på grund av det aktuella tillståndet för en tråd. Till exempel försöker starta om en avbrutna tråden genom att anropa <xref:System.Threading.Thread.Start%2A>utlöser en ThreadStateException på en tråd som har avslutats.</xref:System.Threading.Thread.Start%2A>       ThreadStateException använder HRESULT COR_E_THREADSTATE, som har värdet 0x80131520.       En lista över inledande egenskapsvärden för en instans av ThreadStateException, finns det <xref:System.Threading.ThreadStateException.%23ctor%2A>konstruktorer.</xref:System.Threading.ThreadStateException.%23ctor%2A>"
  example:
  - "The following example demonstrates an error that causes the system to throw a `ThreadStateException`.  \n  \n [!code-cpp[ThreadStEx#1](~/add/codesnippet/cpp/t-system.threading.threa_3_1.cpp)]\n [!code-vb[ThreadStEx#1](~/add/codesnippet/visualbasic/t-system.threading.threa_3_1.vb)]\n [!code-cs[ThreadStEx#1](~/add/codesnippet/csharp/t-system.threading.threa_3_1.cs)]  \n  \n This code produces the following output:  \n  \n```  \nIn main. Attempting to restart myThread.  \n    Working thread...  \n    Caught: Thread is running or terminated. Cannot restart.  \n```"
  syntax:
    content: >-
      [System.Runtime.InteropServices.ComVisible(true)]

      public class ThreadStateException : SystemException
  inheritance:
  - System.Exception
  - System.SystemException
  implements: []
  inheritedMembers: []
  platform:
  - net462
- uid: System.Threading.ThreadStateException.#ctor
  id: '#ctor'
  parent: System.Threading.ThreadStateException
  langs:
  - csharp
  name: ThreadStateException()
  nameWithType: ThreadStateException.ThreadStateException()
  fullName: System.Threading.ThreadStateException.ThreadStateException()
  type: Constructor
  assemblies:
  - mscorlib
  namespace: System.Threading
  summary: "Initierar en ny instans av den <xref href=&quot;System.Threading.ThreadStateException&quot;> </xref> klass med standardegenskaper."
  remarks: "Den här konstruktorn initierar den `Message` egenskapen för den nya instansen till en tillhandahålls av systemet meddelande som anger vilket fel som &quot;den begärda åtgärden inte kan utföras på tråden på grund av det aktuella tillståndet.&quot; Det här meddelandet tar hänsyn till den aktuella systemkulturen.       I följande tabell visas de inledande egenskapsvärdena för en instans av <xref:System.Threading.ThreadStateException>.</xref:System.Threading.ThreadStateException>      | Egenskapen | Värdet |   |--------------|-----------|   | <xref:System.Exception.InnerException%2A>| `null`.|   | <xref:System.Exception.Message%2A>| Den lokaliserade felsträngen meddelande. |</xref:System.Exception.Message%2A> </xref:System.Exception.InnerException%2A>"
  syntax:
    content: public ThreadStateException ();
    parameters: []
  overload: System.Threading.ThreadStateException.#ctor*
  exceptions: []
  platform:
  - net462
- uid: System.Threading.ThreadStateException.#ctor(System.String)
  id: '#ctor(System.String)'
  parent: System.Threading.ThreadStateException
  langs:
  - csharp
  name: ThreadStateException(String)
  nameWithType: ThreadStateException.ThreadStateException(String)
  fullName: System.Threading.ThreadStateException.ThreadStateException(String)
  type: Constructor
  assemblies:
  - mscorlib
  namespace: System.Threading
  summary: "Initierar en ny instans av den <xref href=&quot;System.Threading.ThreadStateException&quot;> </xref> klass med ett angivet meddelande."
  remarks: "Den här konstruktorn initierar den `Message` -egenskapen för en ny instans med hjälp av den `message` parameter.       I följande tabell visas de inledande egenskapsvärdena för en instans av <xref:System.Threading.ThreadStateException>.</xref:System.Threading.ThreadStateException>      | Egenskapen | Värdet |   |--------------|-----------|   | <xref:System.Exception.InnerException%2A>| `null`.|   | <xref:System.Exception.Message%2A>| Meddelandet felsträng. |</xref:System.Exception.Message%2A> </xref:System.Exception.InnerException%2A>"
  syntax:
    content: public ThreadStateException (string message);
    parameters:
    - id: message
      type: System.String
      description: "Felmeddelandet som förklarar orsaken till undantaget."
  overload: System.Threading.ThreadStateException.#ctor*
  exceptions: []
  platform:
  - net462
- uid: System.Threading.ThreadStateException.#ctor(System.Runtime.Serialization.SerializationInfo,System.Runtime.Serialization.StreamingContext)
  id: '#ctor(System.Runtime.Serialization.SerializationInfo,System.Runtime.Serialization.StreamingContext)'
  parent: System.Threading.ThreadStateException
  langs:
  - csharp
  name: ThreadStateException(SerializationInfo,StreamingContext)
  nameWithType: ThreadStateException.ThreadStateException(SerializationInfo,StreamingContext)
  fullName: System.Threading.ThreadStateException.ThreadStateException(SerializationInfo,StreamingContext)
  type: Constructor
  assemblies:
  - mscorlib
  namespace: System.Threading
  summary: "Initierar en ny instans av den <xref href=&quot;System.Threading.ThreadStateException&quot;> </xref> klass med serialiserade data."
  remarks: "Den här konstruktorn anropas under deserialisering till att fylla Undantagsobjektet som överförs via en dataström. Mer information finns i [XML och SOAP-serialiseringen](~/add/includes/ajax-current-ext-md.md)."
  syntax:
    content: protected ThreadStateException (System.Runtime.Serialization.SerializationInfo info, System.Runtime.Serialization.StreamingContext context);
    parameters:
    - id: info
      type: System.Runtime.Serialization.SerializationInfo
      description: "Den <xref href=&quot;System.Runtime.Serialization.SerializationInfo&quot;> </xref> som innehåller det serialiserade objektdata om undantag som genereras."
    - id: context
      type: System.Runtime.Serialization.StreamingContext
      description: "Den <xref:System.Runtime.Serialization.StreamingContext>som innehåller detaljerad information om käll- och målservrar.</xref:System.Runtime.Serialization.StreamingContext>"
  overload: System.Threading.ThreadStateException.#ctor*
  exceptions: []
  platform:
  - net462
- uid: System.Threading.ThreadStateException.#ctor(System.String,System.Exception)
  id: '#ctor(System.String,System.Exception)'
  parent: System.Threading.ThreadStateException
  langs:
  - csharp
  name: ThreadStateException(String,Exception)
  nameWithType: ThreadStateException.ThreadStateException(String,Exception)
  fullName: System.Threading.ThreadStateException.ThreadStateException(String,Exception)
  type: Constructor
  assemblies:
  - mscorlib
  namespace: System.Threading
  summary: "Initierar en ny instans av den <xref href=&quot;System.Threading.ThreadStateException&quot;> </xref> klass med ett angivet meddelande och en referens till det interna undantaget som är orsaken till det här undantaget."
  remarks: "Ett undantag som utlöses när ett direkt resultat av ett tidigare undantag ska innehålla en referens till det tidigare undantaget i den <xref:System.Exception.InnerException%2A>egenskapen.</xref:System.Exception.InnerException%2A> Den <xref:System.Exception.InnerException%2A>-egenskap returnerar samma värde som överförs till konstruktorn eller `null` om den <xref:System.Exception.InnerException%2A>egenskapen tillhandahåller inte värdet i det interna undantaget för konstruktorn.</xref:System.Exception.InnerException%2A> </xref:System.Exception.InnerException%2A>       I följande tabell visas de inledande egenskapsvärdena för en instans av <xref:System.Threading.ThreadStateException>.</xref:System.Threading.ThreadStateException>      | Egenskapen | Värdet |   |--------------|-----------|   | <xref:System.Exception.InnerException%2A>| Referens för inre undantag. |   | <xref:System.Exception.Message%2A>| Meddelandet felsträng. |</xref:System.Exception.Message%2A> </xref:System.Exception.InnerException%2A>"
  syntax:
    content: public ThreadStateException (string message, Exception innerException);
    parameters:
    - id: message
      type: System.String
      description: "Felmeddelandet som förklarar orsaken till undantaget."
    - id: innerException
      type: System.Exception
      description: "Undantag som är orsaken till det aktuella undantaget. Om den `innerException` parametern är inte <xref uid=&quot;langword_csharp_null&quot; name=&quot;null&quot; href=&quot;&quot;> </xref>, det aktuella undantaget utlöses i en <xref uid=&quot;langword_csharp_catch&quot; name=&quot;catch&quot; href=&quot;&quot;> </xref> block som hanterar det interna undantaget."
  overload: System.Threading.ThreadStateException.#ctor*
  exceptions: []
  platform:
  - net462
references:
- uid: System.SystemException
  isExternal: false
  name: System.SystemException
- uid: System.Threading.ThreadStateException.#ctor
  parent: System.Threading.ThreadStateException
  isExternal: false
  name: ThreadStateException()
  nameWithType: ThreadStateException.ThreadStateException()
  fullName: System.Threading.ThreadStateException.ThreadStateException()
- uid: System.Threading.ThreadStateException.#ctor(System.String)
  parent: System.Threading.ThreadStateException
  isExternal: false
  name: ThreadStateException(String)
  nameWithType: ThreadStateException.ThreadStateException(String)
  fullName: System.Threading.ThreadStateException.ThreadStateException(String)
- uid: System.String
  parent: System
  isExternal: true
  name: String
  nameWithType: String
  fullName: System.String
- uid: System.Threading.ThreadStateException.#ctor(System.Runtime.Serialization.SerializationInfo,System.Runtime.Serialization.StreamingContext)
  parent: System.Threading.ThreadStateException
  isExternal: false
  name: ThreadStateException(SerializationInfo,StreamingContext)
  nameWithType: ThreadStateException.ThreadStateException(SerializationInfo,StreamingContext)
  fullName: System.Threading.ThreadStateException.ThreadStateException(SerializationInfo,StreamingContext)
- uid: System.Runtime.Serialization.SerializationInfo
  parent: System.Runtime.Serialization
  isExternal: false
  name: SerializationInfo
  nameWithType: SerializationInfo
  fullName: System.Runtime.Serialization.SerializationInfo
- uid: System.Runtime.Serialization.StreamingContext
  parent: System.Runtime.Serialization
  isExternal: true
  name: StreamingContext
  nameWithType: StreamingContext
  fullName: System.Runtime.Serialization.StreamingContext
- uid: System.Threading.ThreadStateException.#ctor(System.String,System.Exception)
  parent: System.Threading.ThreadStateException
  isExternal: false
  name: ThreadStateException(String,Exception)
  nameWithType: ThreadStateException.ThreadStateException(String,Exception)
  fullName: System.Threading.ThreadStateException.ThreadStateException(String,Exception)
- uid: System.Exception
  parent: System
  isExternal: true
  name: Exception
  nameWithType: Exception
  fullName: System.Exception
- uid: System.Threading.ThreadStateException.#ctor*
  parent: System.Threading.ThreadStateException
  isExternal: false
  name: ThreadStateException
  nameWithType: ThreadStateException.ThreadStateException
